xls/dslx/stdlib/std.x

```diff
@@ -396,6 +396,41 @@ fn round_up_to_nearest_test() {
     assert_eq(u32:4, round_up_to_nearest(u32:4, u32:2));
 }
 
+// Returns `x` rounded up to `y`, where `y` is a positive power of 2
+fn round_up_to_pow2_unsigned<N: u32>(x: uN[N], y: uN[N]) -> uN[N] { (x + y - uN[N]:1) & -y }
+
+#[test]
+fn test_round_up_to_pow2_unsigned() {
+    assert_eq(round_up_to_pow2_unsigned(u16:0, u16:8), u16:0);
+    assert_eq(round_up_to_pow2_unsigned(u16:1, u16:8), u16:8);
+    assert_eq(round_up_to_pow2_unsigned(u16:7, u16:8), u16:8);
+    assert_eq(round_up_to_pow2_unsigned(u16:8, u16:8), u16:8);
+    assert_eq(round_up_to_pow2_unsigned(u16:9, u16:8), u16:16);
```

Comment:
shouldn't this be rounded to `8`?

Reply:
Yes, see https://github.com/google/xls/pull/1238#discussion_r1466484406 for more details.